academy/3-ibc/5-token-transfer.md

```diff
@@ -238,12 +187,103 @@ func (k Keeper) SendTransfer(
 }
 ```
 
+Take a look at the type [definition of a token packet](https://github.com/cosmos/ibc-go/blob/main/proto/ibc/applications/transfer/v2/packet.proto) before diving further into the code:
+
+```protobuf
+syntax = "proto3";
+
+package ibc.applications.transfer.v2;
+
+option go_package = "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types";
+
+// FungibleTokenPacketData defines a struct for the packet payload
+// See FungibleTokenPacketData spec:
+// https://github.com/cosmos/ibc/tree/master/spec/app/ics-020-fungible-token-transfer#data-structures
+message FungibleTokenPacketData {
+    // the token denomination to be transferred
+    string denom = 1;
+    // the token amount to be transferred
+    string amount = 2;
+    // the sender address
+    string sender = 3;
+    // the recipient address on the destination chain
+    string receiver = 4;
+    // optional memo
+    string memo = 5;
+}
+```
+
+<HighlightBox type="note">
+
+Note that an optional _memo_ field was recently added to the packet definition. More details on the motivation, use cases and consequences will follow shortly.
+
+</HighlightBox>
+
+### Receiving a transfer packet
+
+A relayer will then pick up the `SendPacket` event and submit a `MsgRecvPacket` on the destination chain.
+
+This will then trigger an `OnRecvPacket` callback that will decode a packet and apply the transfer token application logic:
+
+```go
+// OnRecvPacket implements the IBCModule interface. A successful acknowledgement
+// is returned if the packet data is successfully decoded and the receive application
+// logic returns without error.
+func (im IBCModule) OnRecvPacket(
```

Comment:
The [latest code](https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/ibc_module.go#L167) has some extra changes, up to you if you would like to have it the same here.